powertools-parameters/src/test/java/software/amazon/lambda/powertools/parameters/SecretsProviderTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.parameters;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+
+import java.util.Base64;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class SecretsProviderTest {
```

Comment:
Some of the  same suggestions as for `SSMProviderTest `

Reply:
Same answer, we must favor unit tests over integration tests (test pyramid). This class tests exactly what she needs to, retrieve the parameter in the 'real' store (which is mocked). If we want integration test, let's do one with a real call a Secrets Manager (maybe with localstack).

Reply:
yes i think we should have one full stack stack showcasing all the public api method calls. As long as we inject mocked client to the provider, we can proceed cover it pretty well